allauth/socialaccount/providers/frontierdev/provider.py

```diff
@@ -0,0 +1,53 @@
+import hashlib
+from urllib.parse import urlencode
+
+from allauth.account.models import EmailAddress
+from allauth.socialaccount.providers.base import ProviderAccount
+from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
+
+
+class FDevAccount(ProviderAccount):
+    def get_profile_url(self):
+        return None
+
+    def get_avatar_url(self):
+        return "https://www.gravatar.com/avatar/%s?%s" % (
+            hashlib.md5(self.account.extra_data.get("email").lower().encode('utf-8')).hexdigest(),
+            urlencode({'d': 'mp'}))
+
+    def to_str(self):
+        dflt = super(FDevAccount, self).to_str()
+        full_name = "%s %s" % (self.account.extra_data.get("firstname", dflt),
+                               self.account.extra_data.get("lastname", dflt))
+        return full_name
+
+
+class FDevProvider(OAuth2Provider):
+    id = "frontierdev"
```

Comment:
Wondering, why is this called "frontierdev" and not just "frontier" or perhaps "frontierstore" ?

Reply:
Historically, it was called Frontier Developments PLC then they shortened it to Frontier. However, their stock market trading abbreviation is FDEV.

[Frontier Developments OAuth2](http://hosting.zaonce.net/docs/oauth2/instructions.html) -- bottom of this page shows Frontier Developments. If you want for consistency, I can refactor the code to Frontier. LMK.

Reply:
I went ahead and refactored the name to frontier.